cunumeric/array.py

```diff
@@ -2810,7 +2902,13 @@ def max(
         )
 
     @add_boilerplate()
-    def mean(self, axis=None, dtype=None, out=None, keepdims=False):
+    def mean(
+        self,
+        axis: Any = None,
+        dtype: Union[np.dtype[Any], None] = None,
```

Comment:
Noting this as future work: Any public functions in `array.py` and `module.py` should accept `npt.DTypeLike`, e.g. they should accept things like `int` and `np.float64`. Internal functions, mostly in `thunk.py`, `eager.py` and `deferred.py` (and sometimes in `array.py` and `module.py`) can be constrained to only `np.ndtype[Any]`, with the frontend functions converting as appropriate.